asserts/repair.go

```diff
@@ -131,6 +153,38 @@ func assembleRepair(assert assertionBase) (Assertion, error) {
 	if err != nil {
 		return nil, err
 	}
+	modes, err := checkStringList(assert.headers, "modes")
+	if err != nil {
+		return nil, err
+	}
+	bases, err := checkStringList(assert.headers, "bases")
+	if err != nil {
+		return nil, err
+	}
+
+	// verify that modes is a list of only "run", "recover" and "install"
+	if len(modes) != 0 {
+		for _, m := range modes {
+			// note we could import boot here to use i.e. boot.ModeRun, but that
+			// might make import cycles happen some day if boot needs to use
+			// the asserts package for some odd reason, so instead just use the
+			// values directly, they're unlikely to change now
+			if !strutil.ListContains([]string{"run", "recover", "install"}, m) {
+				return nil, fmt.Errorf("header \"modes\" contains an invalid element: %q (valid values are run, recover and install)", m)
+			}
+		}
+
+		// if modes is non-empty, then bases must be core2X, i.e. core20+
+
+		// TODO: we should have a more centralized helper for identifying core2X
+		//       snaps to make it easier to add new base snaps
+		uc20AndUpBaseSnapRegexp := regexp.MustCompile(`^core2(0|2)$`)
```

Comment:
this is probably too strict otoh. I think we should do the reverse and instead:

* always check that the content of bases are valid snap names with naming.ValidateSnap(name)
* only error here if bases contains "core" or "core18"

Reply:
ok, I changed it slightly to also error if the base snap name is `core16` too and independently verify all elements of `bases` are valid snap names